### ISSUE TITLE
`webContents` never emits `unresponsive` while `webContents.debugger` is attached

### ISSUE THREAD

**Issue by BenItBuhner** [electron/electron#54323]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

44.4.5

### What operating system(s) are you using?

Ubuntu

### Operating System Version

Ubuntu 24.04 (X11, Xvfb)

### What arch are you using?

x64

### Does the issue also appear in Chromium / Google Chrome?

No

### Expected Behavior

A renderer that stops responding is reported through `webContents.on('unresponsive')` whether or not the app holds a `webContents.debugger` session on that `webContents`. A session that only issues emulation or instrumentation commands (`Emulation.*`, `Network.enable`, …) and is not paused at a breakpoint should not disable the hang report. If the suppression is intended, it should be documented on the `unresponsive` event and on `debugger.attach()`; today neither page mentions it.

### Actual Behavior

While any `webContents.debugger` session is attached, `unresponsive` (and the matching `responsive`) is never emitted for that `webContents`. Detaching restores it.

With the repro below (a button whose click handler runs `for(;;){}`; the click is sent with `sendInputEvent`):

- no debugger attached: `unresponsive` fires 16.0 s after the click;
- `debugger.attach('1.3')` plus one `Emulation.setHardwareConcurrencyOverride` command, session left attached: no `unresponsive` within 60 s (`debugger.isAttached()` is `true`);
- same attach and command, then `debugger.detach()` before the click: `unresponsive` fires 16.0 s after the click.

Output on 44.4.5 (Ubuntu 24.04, Xvfb, sandbox enabled):

```
A: no debugger
[+   636ms] clicked hang button
[+ 16644ms] EVENT unresponsive

B: debugger attached
[+   132ms] debugger attached; Emulation.setHardwareConcurrencyOverride -> {}
[+   634ms] clicked hang button
[+ 60695ms] RESULT variant=B waited=60s after click     (no EVENT line)

C: attached, then detached before the click
[+   643ms] clicked hang button
[+ 16651ms] EVENT unresponsive
```

Same on 45.0.0-alpha.11 (Chromium 155.0.8046.0): A fires at 16.0 s, B never fires within 60 s.

Any app that keeps a CDP session on a page for its own purposes (emulation, network instrumentation, throttling) therefore loses hang detection for that page without any indication. The same happens to every page under Playwright for Electron or any `--remote-debugging-port` client with `Target.setAutoAttach`, so hang handling cannot be tested under such a harness.

### Additional Information

Minimal repro, `main.js` (with `{"main":"main.js"}` in `package.json`). Run `electron . A 45`, `electron . B 60`, `electron . C 45` (A = no debugger, B = attached, C = attached then detached; the number is how many seconds to wait after the click). The renderer is deliberately hung; the process exits by itself after the wait.

```js
const { app, BrowserWindow } = require('electron')

const variant = process.argv.find((a) => /^[ABC]$/.test(a)) || 'A'
const waitMs = Number(process.argv.find((a) => /^\d+$/.test(a)) || 45) * 1000
const t0 = Date.now()
const log = (...args) => console.log(`[+${String(Date.now() - t0).padStart(6)}ms]`, ...args)

app.whenReady().then(async () => {
  log('versions', process.versions.electron, process.versions.chrome)
  const win = new BrowserWindow({ width: 800, height: 600 })
  const wc = win.webContents
  wc.on('unresponsive', () => log('EVENT unresponsive'))
  wc.on('responsive', () => log('EVENT responsive'))
  await wc.loadURL('data:text/html,<button style="position:fixed;left:0;top:0;width:200px;height:100px" onclick="for(;;){}">hang</button>')

  if (variant === 'B' || variant === 'C') {
    wc.debugger.attach('1.3')
    const r = await wc.debugger.sendCommand('Emulation.setHardwareConcurrencyOverride', { hardwareConcurrency: 2 })
    log('debugger attached; Emulation.setHardwareConcurrencyOverride ->', JSON.stringify(r))
  }
  if (variant === 'C') {
    wc.debugger.detach()
    log('debugger detached')
  }
  log('debugger.isAttached() =', wc.debugger.isAttached())

  await new Promise((r) => setTimeout(r, 500))
  wc.sendInputEvent({ type: 'mouseDown', x: 50, y: 50, button: 'left', clickCount: 1 })
  wc.sendInputEvent({ type: 'mouseUp', x: 50, y: 50, button: 'left', clickCount: 1 })
  log('clicked hang button')

  setTimeout(() => {
    log(`RESULT variant=${variant} waited=${waitMs / 1000}s after click (no EVENT line above = never fired)`)
    process.exit(0)
  }, waitMs)
})
```

Environment used: Electron 44.4.5 (Chromium 152.0.7977.130, Node 24.21.0) and 45.0.0-alpha.11 (Chromium 155.0.8046.0), Ubuntu 24.04.4 x64 under Xvfb (X11), Chromium sandbox enabled.

<details><summary>In-depth investigation</summary>
<p>

The gate is applied in Electron's code. `patches/chromium/refactor_unfilter_unresponsive_events.patch` (from #44629) removes Chromium's `ShouldIgnoreUnresponsiveRenderer()` early return from `WebContentsImpl::RendererUnresponsive` so that every hang reaches Electron, and Electron re-applies the same predicate before emitting:

- `main`: `shell/browser/api/electron_api_web_contents.cc`, `WebContents::RendererUnresponsive`: `bool should_ignore = web_contents_impl->ShouldIgnoreUnresponsiveRenderer(); … if (!should_ignore && visible && renderer_initialized) EmitWithoutEvent("unresponsive", event_object);`
- `44-x-y`: the same three fields are put on the internal `-unresponsive` event and `lib/browser/api/web-contents.ts` emits `unresponsive` only when `!event.shouldIgnore && event.visible && event.rendererInitialized`.

`content::WebContentsImpl::ShouldIgnoreUnresponsiveRenderer()` returns `DevToolsAgentHost::IsDebuggerAttached(this)` as its last step, i.e. true for any attached DevTools client of the `WebContents`, with the upstream comment `// TODO(pfeldman): Fix this to only return true if the renderer is *actually* sitting on a breakpoint. https://crbug.com/684202`. `webContents.debugger.attach()` creates such a client, so `shouldIgnore` is true for as long as the session lives. Chrome uses the predicate only to hide its hung-page dialog; in Electron it silently removes the app's only hang signal.

Possible directions: emit `unresponsive` regardless of `IsDebuggerAttached` (the visibility and sleep cases were already reworked in #53529, so the predicate is already Electron's decision); or track whether the attached session is actually paused (`Debugger.paused` without a matching `Debugger.resumed`) and ignore only then; or exclude `webContents.debugger` sessions from the check (they belong to the app, not to a developer at a breakpoint). At minimum, document the current behaviour on `unresponsive` and `debugger.attach()`.

See also #14257 (2018, the inverse case: `unresponsive` firing while paused in DevTools, which is what the debugger check protects against).

</p>
</details>

Found while building Zenium, an Electron-based browser: https://github.com/BenItBuhner/Zenium

Prepared with AI assistance (Claude, via Cursor): the investigation and write-up were AI-assisted; the repro output above was produced by running the code as shown.
